### DISCUSSION TITLE
Focus plugin .noscroll modifier iOS issues

### DISCUSSION THREAD

**Discussion by tomblanchard** [alpinejs/alpine#3037] [7. Bugs (archived)]
It seems that the focus plugin .noscroll modifier no longer prevents body scrolling on iOS - I think this is a new bug from one of the recent 15.x iOS updates because it previously worked flawlessly.

Here's a screencast taken from the Alpine docs example on latest iOS version (15.5):

https://user-images.githubusercontent.com/3143764/178741239-1231221a-31b2-424e-8fc8-7c6e02bb67f3.mov

I've looked into this and it's related to the [disableScrolling](https://github.com/alpinejs/alpine/blob/main/packages/focus/src/index.js#L185) function, it's strategy to prevent body scrolling is to set `html { overflow: hidden; }` - this seems to work everywhere except iOS. Perhaps it's necessary to outsource this to a dedicated library such as [body-scroll-lock](https://github.com/willmcpo/body-scroll-lock)? At a bare minimum, it'd be good to have the ability to set our own disableScrolling function so we have the ability to override the default.

**Comment by danddanddand** [comment]
see https://github.com/alpinejs/alpine/issues/3038#issuecomment-1183273916
